homeassistant/components/shelly/utils.py

```diff
@@ -571,3 +574,15 @@ def async_remove_orphaned_virtual_entities(
 
     if orphaned_entities:
         async_remove_shelly_rpc_entities(hass, platform, mac, orphaned_entities)
+
+
+def get_rpc_ws_url(hass: HomeAssistant) -> str | None:
+    """Return the RPC websocket URL."""
+    try:
+        raw_url = get_url(hass, prefer_external=False, allow_cloud=False)
+    except NoURLAvailableError:
+        LOGGER.debug("URL not available, skipping outbound websocket setup")
```

Comment:
I would vote to move to warning and point the user to the documentation for manual setup

Reply:
I started with debug here to keep it simple, but I think it would be good to make a repair issue in a future PR